src/components/application_manager/src/commands/command_request_impl.cc

```diff
@@ -105,20 +108,45 @@ const std::string CreateInfoForUnsupportedResult(
 }
 
 bool CheckResultCode(const ResponseInfo& first, const ResponseInfo& second) {
+  if (!first.is_ok && !second.is_ok) {
+    return false;
+  }
+
+  if (!first.is_ok && second.is_unsupported_resource) {
+    return false;
+  }
+
+  if (!second.is_ok && first.is_unsupported_resource) {
+    return false;
+  }
   if (first.is_ok && second.is_unsupported_resource &&
-      second.interface_state == HmiInterfaces::STATE_NOT_AVAILABLE) {
+      HmiInterfaces::STATE_NOT_AVAILABLE != second.interface_state) {
     return true;
   }
+  if (first.is_ok && second.is_invalid_enum) {
+    return true;
+  }
+
+  if (second.is_ok && first.is_invalid_enum) {
+    return true;
+  }
+
   return false;
 }
 
-bool IsResultCodeUnsupported(const ResponseInfo& first,
-                             const ResponseInfo& second) {
-  return ((first.is_ok || first.is_invalid_enum) &&
-          second.is_unsupported_resource) ||
-         ((second.is_ok || second.is_invalid_enum) &&
-          first.is_unsupported_resource) ||
-         (first.is_unsupported_resource && second.is_unsupported_resource);
+bool IsResultCodeWarning(const ResponseInfo& first,
+                         const ResponseInfo& second) {
+  const bool first_is_ok_second_is_warn =
+      (first.is_ok || first.is_invalid_enum) &&
+      hmi_apis::Common_Result::WARNINGS == second.result_code;
+  const bool second_is_ok_first_is_warn =
```

Comment:
Should remove duplicate comparisons

Reply:
@jacobkeeler checks reworked to avoid double checks.